### ISSUE TITLE
What needs to be done?

### ISSUE THREAD

**Issue by Eh2406** [aldanor/hdf5-rs#17]
What is left on the todo list to get this useable? What can developers with different backgrounds do to help?

**Comment by aldanor** [comment]
Hi @Eh2406, apologies, I haven't had much time recently due to heavy involvement with a few other projects, but it's been on my mind to bring this to usable state (especially with the fact that rust-ndarray reached a fairly useful state lately) -- I've mentioned a few things in #14. One question is how to deal with dataset datatypes (mentioned there).

I've just pushed an existing branch I've had, with the latest work -- https://github.com/aldanor/hdf5-rs/commit/4429b007c4c7e4f0dfb75b6e9eb555e349593b3e -- changing trait-based inheritance to specializations; while it's a lot cleaner code-wise and doesn't require end-user to import a million different traits, almost **all** documentation basically just disappears (due to issues in rustdoc mentioned in #14). (would also appreciate if anyone voiced a thought on whether specializations are indeed better than a multitude of traits; to me personally it looks a lot better)

Fixing the aforementioned rustdoc issue or somehow helping the core team with that would probably be the *greatest* help of all; or figuring and implementing an alternative way (like in [rust-xcb](https://github.com/rtbo/rust-xcb)) to generate the docs give there's a lot of aliases and specializations.

Cheers!

**Comment by Eh2406** [comment]
I spent some time trying to use the new branch, and I can confirm that it is not easy to grock from the docs as they are. :-P Perhaps some examples reading and writing with different types could help paper over what rust-doc refuses to render and give some context on how the parts fit together. Some of the confusion is my background, I have only used hdf's from pandas, so if the concepts map to the c-api then I'd miss the connection. Perhaps some links to comparable parts of other languages api's could help?

What practically would be involved in doing the docs the way rust-xcb does it? I looked at the the xcb project but could not find exactly what you are referring to. I also wonder if the new [include external files in documentation (RFC 1990)](https://github.com/rust-lang/rust/pull/44781) could be used to cover this?
